numba/cuda/tests/cudapy/test_dispatcher.py

```diff
@@ -288,5 +290,33 @@ def add_device(a, b):
         self.assertEqual("Add two integers, device version", add_device.__doc__)
 
 
+@skip_on_cudasim('Dispatcher objects not used in the simulator')
+class TestDispatcherDeprecation(CUDATestCase):
+    def check_warning(self, warnings, expected_str, category):
+        self.assertEqual(len(warnings), 3)
+        for w in warnings:
+            self.assertEqual(w.category, category)
+            self.assertIn(expected_str, str(w.message))
+
+    def test_ptx_deprecation(self):
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
This probably needs to call: https://github.com/numba/numba/blob/53ea89fee1a19471ce1b1907f9f3d98acd1b6730/numba/tests/support.py#L141 so as to not break if any of the internal, and safe to ignore, warnings are raised.